Dom/gemini system prompt switching (#1260)
pipecat-ai/pipecat: examples/phone-chatbot/README.md

     -d '{"dialoutNumber": "+18057145330", "detectVoicemail": true}'
```

### New! Using Gemini 2.0 Flash Lite with Daily

We have introduced support for Google's Gemini 2.0 Flash Lite model in this example. This lightweight model offers faster response times and reduced costs while maintaining good conversational capabilities.

**Quick Start**
To use the Gemini-based bot instead of OpenAI:

```shell
curl -X POST "http://localhost:7860/daily_gemini_start_bot" \                                                                                                        py pipecat
     -H "Content-Type: application/json" \
     -d '{"detectVoicemail": true}'
```

All request body parameters supported by /daily_start_bot (such as detectVoicemail, dialoutNumber, etc.) are also compatible with /daily_gemini_start_bot.

> This example uses context switching to help steer the bot in the right direction. As Flash Lite is a smaller model, getting it to consistently call functions was difficult for these longer prompts. Breaking the prompt

Reviewer: It may make more sense to just state what the best practice is.

```suggestion
This example uses context switching to help steer the bot in the right direction. As Flash Lite is a smaller model, breaking the prompt down into smaller piece helps to improve the bot's accuracy.
```

Also, what do you mean by smaller prompts? Maybe a short example would help to explain this line.

down into smaller pieces helped improve the accuracy of the bot.

**Implementation Details**
The implementation is available in bot_daily_gemini.py and features:

> Staged prompting approach: Breaking down complex tasks into smaller, more focused prompts to improve the lightweight model's performance

Reviewer: Maybe make this a list?

```suggestion
- Staged prompting approach: Breaking down complex tasks into smaller, more focused prompts to improve the lightweight model's performance
```

Dynamic context switching: The bot can change its behavior in real-time based on what it detects (voicemail vs. human caller)
Function-based architecture: Uses function calling to trigger context switches and call termination

**Optimizations for Lightweight Models**
Working with Gemini 2.0 Flash Lite required some specific optimizations:

> Simplified prompts: Each prompt focuses on a single task with clear instructions

Reviewer: Same here, let's make it a list.
```suggestion
- Simplified prompts: Each prompt focuses on a single task with clear instructions
```

Function-driven state changes: The model calls specific functions to switch between different conversation modes
Reduced context requirements: Each stage maintains only the context needed for its specific purpose

> This approach significantly improves the consistency of function calling in this lightweight model, which was challenging with longer, more complex prompts.

Reviewer: Delete this too?
